subsys/net/lib/openthread/platform/entropy.c

```diff
@@ -15,12 +16,9 @@
 
 LOG_MODULE_REGISTER(net_otPlat_entropy, CONFIG_OPENTHREAD_L2_LOG_LEVEL);
 
-#if !defined(CONFIG_ENTROPY_HAS_DRIVER)
-#error OpenThread requires an entropy source for a TRNG
```

Comment:
I'd degrade it to a warning level, but still print out a warning in case we don't have a valid entorpy device in the system, something like:
```
"No entropy device in the system, cryptographic operations may be insecure!"
```
Just that the user is aware it's using a fake entropy source.